boptimol/molecule.py

```diff
@@ -75,24 +81,38 @@ def __init__(self, filename: str, calculator: Calculator = None):
         #  .. geometry with a trust range
 
         # TODO: adjust this based on bond elements / types
-        degrees_of_freedom = self.end_angles + len(self.dihedrals)
-
-        lower_bounds = np.zeros(degrees_of_freedom)
-        upper_bounds = np.zeros(degrees_of_freedom)
+        self.degrees_of_freedom = self.end_angles + len(self.dihedrals)
 
+        lower_bounds = np.zeros(self.degrees_of_freedom)
+        upper_bounds = np.zeros(self.degrees_of_freedom)
+        
+        '''chemcoord helper functions'''
+        #getElements in a bond
+        getElement = lambda idx: (self.zmat['atom'][idx], 
```

Comment:
Do you want to make this a full method (e.g., getBondElements() returns a tuple?) .. it seems like it would be useful elsewhere.

Reply:
These functions were much used right now, so thats why a created a lambda function. I will actually make these changes in chemcoord and make a pull request.

Reply:
In the meantime, as I mentioned, I'd put this into the `Molecule` class - you will definitely use it in the initial guess code.